api/client/client_test.go

```diff
@@ -136,6 +137,38 @@ func TestClient_NewCollection(t *testing.T) {
 	})
 }
 
+func TestIt(t *testing.T) {
+	type Owner struct {
+		ID       core.InstanceID `json:"_id"`
+		ThreadID thread.ID       `json:"threadId"`
+	}
+	collectionName := "owners"
+
+	client, done := setup(t)
+	defer done()
+
+	id := thread.NewIDV1(thread.Raw, 32)
+	err := client.NewDB(context.Background(), id)
+	checkErr(t, err)
+	sch := util.SchemaFromInstance(Owner{}, false)
+	schBytes, err := json.MarshalIndent(sch, "", "  ")
+	checkErr(t, err)
+	schStr := string(schBytes)
+	t.Logf("\n%v", schStr)
```

Comment:
Here is the output of this log statement, displaying the json schema string:
```
        {
          "$schema": "http://json-schema.org/draft-04/schema#",
          "$ref": "#/definitions/Owner",
          "definitions": {
            "ID": {
              "additionalProperties": false,
              "type": "object"
            },
            "Owner": {
              "required": [
                "_id",
                "threadId"
              ],
              "properties": {
                "_id": {
                  "type": "string"
                },
                "threadId": {
                  "$schema": "http://json-schema.org/draft-04/schema#",
                  "$ref": "#/definitions/ID"
                }
              },
              "additionalProperties": false,
              "type": "object"
            }
          }
        }
```
The `ID` type that represents thread id looks strange. It is an `object` but has no properties, presumably because the `str` property of `thread.ID` is internal/private.